website/views.py

```diff
@@ -38,53 +85,147 @@ def register(request):
             messages.success(request, "Successfully Registered")
             return redirect('home')
         else:
-            return render(request, 'register.html', {'form': form})
-    else:
+            return render(request, self.template_name, {'form': form})
+    def get(self,request):
         form = SignUpForm()
-        return render(request, 'register.html', {'form': form})
-    return render(request, 'register.html', {'form': form})
+        return render(request, self.template_name, {'form': form})
 
+# Function Based Record View
+# def record(request,pk):
+#     if request.user.is_authenticated:
+#         record=Record.objects.get(id=pk)
+#         return render(request,'record.html',{'cust_record':record})
+#     else:
+#         messages.success(request,"First Login or signup please !!!")
+#         return redirect('home')
 
-def record(request,pk):
-    if request.user.is_authenticated:
-        record=Record.objects.get(id=pk)
-        return render(request,'record.html',{'cust_record':record})
-    else:
-        messages.success(request,"First Login or signup please !!!")
-        return redirect('home')
+class record(LoginRequiredMixin, View):
+    template_name='record.html'
+    def get_record(pk):
+        try:
+            return Record.objects.get(id=pk)
+        except:
+            return None
+    def get(self,request,pk):
+        rec= record.get_record(pk)
+        if rec:
+            return render(request, self.template_name, {'cust_record': rec})
+        else:
+            messages.success(request,"No Record Found !!!")
+            return redirect('home')
 
-def delete(request,pk):
-    if request.user.is_authenticated:
-        record=Record.objects.get(id=pk)
-        record.delete()
-        messages.success(request,"Record Deleted Successfully !!!")
+    def handle_no_permission(self):
+        # Custom error message when user is not logged in
+        messages.error(self.request, "You need to log in to access this page.")
         return redirect('home')
-    else:
-        messages.success(request,"First Login or signup please !!!")
+
+# Function Based Delete View
+# def delete(request,pk):
+#     if request.user.is_authenticated:
+#         Record.objects.filter(id=pk).delete()
+#         messages.success(request,"Record Deleted Successfully !!!")
+#         return redirect('home')
+#     else:
+#         messages.success(request,"First Login or signup please !!!")
+#         return redirect('home')
+
+class delete(LoginRequiredMixin,View):
+    def del_record(self,pk):
+        try:
+            return Record.objects.filter(id=pk).delete()
+        except Record.DoesNotExist:
+            return None
+    def get(self,request,pk):
+        status=self.del_record(pk)
+        if status[0]>0:
```

Comment:
i guess:
`if status:` will meet the required case here

Reply:
it was not working it should work as it returns the count of objects deleted but when i replaced status[0]>0 with if status it then giving me desired output